packages/snaps-ui/src/validation.ts

```diff
@@ -25,3 +25,72 @@ export function isComponent(value: unknown): value is Component {
 export function assertIsComponent(value: unknown): asserts value is Component {
   assertStruct(value, ComponentStruct, 'Invalid component');
 }
+
+const LINK_REGEX = /(?<protocol>[a-z]+:\/?\/?)(?<host>\S+?(?:\.[a-z]+)+)/giu;
+
+enum AlloweProtocols {
+  Https = 'https:',
+  Mailto = 'mailto:',
+}
+
+const ALLOWED_PROTOCOLS = Object.values(AlloweProtocols);
```

Comment:
Seems easier to just use an array here. That also removes the need for the type cast further on.
```suggestion
const ALLOWED_PROTOCOLS = [
  'https:',
  'mailto:',
];
```